storage/client_raft_test.go

```diff
@@ -813,3 +815,53 @@ func TestRangeDescriptorSnapshotRace(t *testing.T) {
 		}
 	}
 }
+
+// TestReplicateAddAndRemoveRemove verifies multiraft.state remove the
+// remote node correctly when range was removed from the store.
+func TestReplicateAddAndRemoveRemove(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	mtc := startMultiTestContext(t, 3)
+	defer mtc.Stop()
+
+	// Make the split.
+	store0 := mtc.stores[0]
+	splitArgs := adminSplitArgs(proto.KeyMin, []byte("b"), proto.RaftID(1), store0.StoreID())
+	if _, err := store0.ExecuteCmd(context.Background(), &splitArgs); err != nil {
+		t.Fatal(err)
+	}
+
+	raftID := proto.RaftID(2)
+	mtc.replicateRange(raftID, 0, 1, 2)
+
+	mtc.unreplicateRange(raftID, 0, 2)
+	mtc.unreplicateRange(raftID, 0, 1)
+
+	rng, err := mtc.stores[1].GetRange(raftID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// If the remove range action happen before range apply replica config change, the group
+	// will be created again when multiraft received a MsgApp.
+	if err := util.IsTrueWithin(func() bool {
+		if len(rng.Desc().Replicas) == 1 {
```

Comment:
@tamird Yes, wait the remove replica command launched  by```mtc.unreplicateRange(raftID, 0, 1)``` has applied on ```mtc.stores[1]```.  after the command has applied on store1, ```rng``` only has one replica which NodeID=1, StoreID=1.